packages/taiko-client/driver/anchor_tx_constructor/anchor_tx_constructor.go

```diff
@@ -69,6 +69,28 @@ func (c *AnchorTxConstructor) AssembleAnchorTx(
 	return c.rpc.TaikoL2.Anchor(opts, l1Hash, l1Header.Root, l1Height.Uint64(), uint32(parentGasUsed))
 }
 
+func (c *AnchorTxConstructor) AssembleNullAnchorTx(
+	ctx context.Context,
+	// Height of the L2 block which including the TaikoL2.anchor transaction.
+	l2Height *big.Int,
+	baseFee *big.Int,
+	parentGasUsed uint64,
+) (*types.Transaction, error) {
+	opts, err := c.transactOpts(ctx, l2Height, baseFee)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Info(
+		"Null anchor arguments",
+		"l2Height", l2Height,
+		"baseFee", utils.WeiToGWei(baseFee),
+		"gasUsed", parentGasUsed,
+	)
+
+	return c.rpc.TaikoL2.Anchor(opts, common.BigToHash(common.Big0), common.BigToHash(common.Big0), common.Big0.Uint64(), uint32(parentGasUsed))
```

Comment:
Setting basefee here can be taken from L1 at the current time instead of setting itto 0. 0 baseFee and 0 gas used could lead to problems. these can be simulated using `eth_call` and `eth_estimateGas` on l1 execution client

Reply:
It's not a 0,  it's taken from the parent, which is the latest L2 block: https://github.com/NethermindEth/taiko-mono/pull/2/files#diff-6d16849826f000a80f50a99715c4b4c162bbe2f7f75cc43f4e4ba6a15e60f708R434

Reply:
That makes no sense, though, because gasUsee in this block will definitely be different and can easily be extracted. 
Let's do `GasUsed`: get it from the l1 call with `eth_estimateGas`
And `BaseFee`: Get it from parent block

Reply:
As I call `eth_estimateGas` I need to specify `ethereum.CallMsg`. I'm not sure for which L1 message you want to estimate gas?

Reply:
Same as the tx that 2ill be sent to mev boost for force inclusion

Reply:
Send the gasUseed value with the payload from the avs node

Reply:
OK, so I understand to have another parameter for the used gas in the `MoveTheHead` method, and I'll pass there the value of used gas in Rust AVS node part.

Reply:
Yes sir